src/macros.rs

```diff
@@ -312,4 +312,16 @@ macro_rules! as_rust {
                     $data_type_option.id)))
         }
     );
+    ($data_type_option:ident, $data_value:ident, Timespec) => (
+        match $data_type_option.id {
+            ColType::Timestamp => {
+                decode_timestamp($data_value.as_slice())
+                    .map(|ts| Timespec::new(ts / 1_000, (ts % 1_000 * 1_000_000) as i32))
```

Comment:
Only the remainder of the division by 1000 (0-999) is multiplied by 1_000_000, so I don't think `checked_mul` is necessary.